### ISSUE TITLE
Issue with third step

### ISSUE THREAD

**Issue by Maelstrom602** [juzeon/poe-openai-proxy#8]
vim config.toml doesn't seem to be a valid command in powershell, command prompt, or python. I tried installing Vim but I couldn't figure out how to use it. How do I make this config.toml file?

**Comment by juzeon** [comment]
Hello. If you are using Windows, you can use text editors like notepad++ or sublime text to edit the configuration. It's not necessary to use vim in Windows.

**Comment by Maelstrom602** [comment]
Oh, I used notepad to put down the various lines- but for some reason I wasn't able to make it a toml file. Currently using Windows 11.

I'm a bit of a newbie.

**Comment by juzeon** [comment]
You can enable the display of file extension first, and then create a txt file and rename it to toml. Google for 'show file extension windows 11'.

**Comment by Maelstrom602** [comment]
Changed it to tmol file. Fired up the python external/api.py command.

FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\xxxx\\poe-openai-proxy\\external\\..\\config.toml'

**Comment by juzeon** [comment]
This means that the program cannot find the configuration file. Did you put it in the right folder (in the same folder as go.mod) and ensure the file name is correct?

**Comment by Maelstrom602** [comment]
I think I got it running. Must've typed in the file name wrong.

 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://192.168.1.238:5000
INFO:werkzeug:Press CTRL+C to quit

Do I still need to build the go backend?

**Comment by juzeon** [comment]
Yes you need. The go backend will call apis of the python backend.

**Comment by Maelstrom602** [comment]
"'chmod' is not recognized as an internal or external command,
operable program or batch file."

**Comment by juzeon** [comment]
On Windows you don't need the chmod step.

**Comment by Maelstrom602** [comment]
Oh damn. Which parts do I keep in? It says go isn't recognized as a command.

**Comment by Maelstrom602** [comment]
Thanks for your patience btw. I sometimes have trouble interpreting instructions.

**Comment by juzeon** [comment]
Never mind. You should install golang the programming language first.

**Comment by Maelstrom602** [comment]
Alright, I resolved step 5. Will I be able to recieve an api at this point? Or do I have to fetch one for the curl command.

**Comment by juzeon** [comment]
Something like 'register... ok' in the output of the go backend should indicate the successful start of the program. After that you can use curl to do a test referring to the openai document or connect other apps to it by changing the openai endpoint of the corresponding app.

**Comment by Maelstrom602** [comment]
It says it's running, but I didn't get "register...ok"

Tried to redo the go build command and it didn't work for some reason.

**Comment by juzeon** [comment]
Did you fill in your poe tokens? The tokens written in the README are for demostration purposes only and invalid.

**Comment by Maelstrom602** [comment]
Yes. Though I could only fill in one. I removed the other. Couldn't get two separate tokens.

**Comment by juzeon** [comment]
Could you provide the full output of the python and the go backend?

**Comment by Maelstrom602** [comment]
C:\Users\xxxx\poe-openai-proxy>python external/api.py # Running on port 5000
 * Serving Flask app 'api'
 * Debug mode: off
INFO:werkzeug:WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on http://127.0.0.1:5000
 * Running on http://192.168.1.238:5000
INFO:werkzeug:Press CTRL+C to quit
--------
main module (test3) does not contain package test3/poe-openai-proxy

**Comment by juzeon** [comment]
Seems that the go program cannot be compile normally. Are this error `main module (test3) does not contain package test3/poe-openai-proxy` occured when you ran `go build`? The code do not have a package named `test3`. Weired. Try to run `go mod tidy` and re-run `go build`.

**Comment by Maelstrom602** [comment]
Ran -go mod tidy- and got this:

"test3/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7: import path "test3/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7" should not have @version
test3/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/examples: import path "test3/go/pkg/mod/github.com/op/go-logging@v0.0.0-20160315200505-970db520ece7/examples" should not have @version"

**Comment by juzeon** [comment]
I've not seen this type of error before. Please do a git pull as I have just commited a new `go mod` file. Also could you please run `go version` and provide the version of your go installation? 

If this does not help, try this pre-compiled executable file: `https://wormhole.app/0WJMP#wMVd6O-jIxQYgy9CSyTujA` (for Windows x86_64, downloadable in 24 hours)
